# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/ProtocolUtils.cpp
# Commit: 7b7819250f79
# Full Hash: 7b7819250f7965227783233af6de0e4e1b789af6
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-23 09:53:24
# Regressor Bug: 1706374
# File Overlap Count: 1
# Description:
#   Bug 1706374 - Part 11: Add NodeController component bridging IPC and Ports, r=handyman
#   
#   The NodeController and NodeChannel types act as the backbone connecting the
#   existing IPC logic and driving the ports routing code. Individual NodeChannel
#   objects wrap and respond to messages from IPC::Channel, and the NodeController
# ==============================================================================

diff -r 8bdac5eedf78 -r 7b7819250f79 ipc/glue/ProtocolUtils.cpp
--- a/ipc/glue/ProtocolUtils.cpp	Tue Jun 22 18:17:21 2021 +0000
+++ b/ipc/glue/ProtocolUtils.cpp	Tue Jun 22 18:17:22 2021 +0000
@@ -639,6 +639,11 @@
   mOtherPid = aOtherPid;
 }
 
+bool IToplevelProtocol::Open(ScopedPort aPort, base::ProcessId aOtherPid) {
+  SetOtherProcessId(aOtherPid);
+  return GetIPCChannel()->Open(std::move(aPort), mSide);
+}
+
 bool IToplevelProtocol::Open(UniquePtr<Transport> aTransport,
                              base::ProcessId aOtherPid, MessageLoop* aThread,
                              mozilla::ipc::Side aSide) {